rust/connlib/tunnel/src/sockets.rs

```diff
@@ -150,3 +145,103 @@ where
         None
     }
 }
+
+struct ThreadedUdpSocket {
+    outbound_tx: flume::r#async::SendSink<'static, DatagramOut>,
+    inbound_rx: flume::r#async::RecvStream<'static, DatagramSegmentIter>,
+}
+
+impl ThreadedUdpSocket {
+    #[expect(clippy::unwrap_in_result, reason = "We unwrap in the new thread.")]
+    fn new(sf: &dyn SocketFactory<UdpSocket>, addr: SocketAddr) -> io::Result<Self> {
+        let mut socket = sf(&addr)?;
```

Comment:
This probably suffers from the same issue as the TUN thread where this is bound to the wrong tokio runtime.